common/src/main/java/org/opensearch/ml/common/MLConfig.java

```diff
@@ -28,43 +29,74 @@ public class MLConfig implements ToXContentObject, Writeable {
 
     public static final String TYPE_FIELD = "type";
 
-    public static final String CONFIG_TYPE_FIELD = "config_type";
-
     public static final String CONFIGURATION_FIELD = "configuration";
 
-    public static final String ML_CONFIGURATION_FIELD = "ml_configuration";
-
     public static final String CREATE_TIME_FIELD = "create_time";
     public static final String LAST_UPDATE_TIME_FIELD = "last_update_time";
 
+    // Adding below three new fields since the original fields, type, configuration, and last_update_time
+    // are not created with correct data types in config index due to missing schema version bump.
+    // Starting 2.15, it is suggested that below fields be used for creating new documents in config index
+
+    public static final String CONFIG_TYPE_FIELD = "config_type";
+
+    public static final String ML_CONFIGURATION_FIELD = "ml_configuration";
+
     public static final String LAST_UPDATED_TIME_FIELD = "last_updated_time";
 
+    private static final Version MINIMAL_SUPPORTED_VERSION_FOR_NEW_CONFIG_FIELDS = CommonValue.VERSION_2_15_0;
+
     @Setter
     private String type;
 
+    @Setter
+    private String configType;
+
     private Configuration configuration;
+    private Configuration mlConfiguration;
     private final Instant createTime;
     private Instant lastUpdateTime;
+    private Instant lastUpdatedTime;
 
     @Builder(toBuilder = true)
-    public MLConfig(String type, Configuration configuration, Instant createTime, Instant lastUpdateTime) {
+    public MLConfig(
+        String type,
+        String configType,
+        Configuration configuration,
+        Configuration mlConfiguration,
+        Instant createTime,
+        Instant lastUpdateTime,
+        Instant lastUpdatedTime
+    ) {
         this.type = type;
+        this.configType = configType;
         this.configuration = configuration;
+        this.mlConfiguration = mlConfiguration;
         this.createTime = createTime;
         this.lastUpdateTime = lastUpdateTime;
+        this.lastUpdatedTime = lastUpdatedTime;
     }
 
     public MLConfig(StreamInput input) throws IOException {
+        Version streamInputVersion = input.getVersion();
```

Comment:
Shouldn't this be optional since it was not there in previous versions for upgrades?